commit 738b3ca3a2afc6b0123ccc5a7a3dab1b9ad054b7 (refs/changes/73/2973/6)
Author: Julien Gilli <julien.gilli@joyent.com>
Date:   2017-11-23T19:19:26+00:00 (1 year, 10 months ago)
    
    VOLAPI-92 nightly2-020-test-system failing in node-sdc-clients volapi tests
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/lib/cli/do_nfs_volumes.js b/lib/cli/do_nfs_volumes.js
index 4bd8bd0..fcbd9cd 100644
--- a/lib/cli/do_nfs_volumes.js
+++ b/lib/cli/do_nfs_volumes.js
@@ -297,6 +297,28 @@ function do_nfs_volumes(subcmd, opts, args, cb) {
     }
 
     vasync.pipeline({arg: context, funcs: [
+        function checkFabricsEnabled(ctx, next) {
+            var err;
+
+            assert.object(self.sdcadm.sdc.metadata, 'self.sdcadm.sdc.metadata');
+
+            /*
+             * Disabling any NFS volumes feature flag does have any requirement,
+             * let alone on support for fabric networks.
+             */
+            if (opts.disable) {
+                next();
+                return;
+            }
+
+            if (!self.sdcadm.sdc.metadata.fabric_cfg) {
+                err = new Error('cannot enable NFS volumes feature: this DC ' +
+                    'is not setup for fabric networks');
+            }
+
+            next(err);
+        },
+
         function checkFeatureDeps(ctx, next) {
             var err;
             var missingFeatureDeps;
diff --git a/lib/post-setup/volapi.js b/lib/post-setup/volapi.js
index 06ca821..504c996 100644
--- a/lib/post-setup/volapi.js
+++ b/lib/post-setup/volapi.js
@@ -168,6 +168,19 @@ function do_volapi(subcmd, opts, args, cb) {
     vasync.pipeline({arg: context, funcs: [
         steps.sapiAssertFullMode,
 
+        function checkFabricsEnabled(ctx, next) {
+            var err;
+
+            assert.object(self.sdcadm.sdc.metadata, 'self.sdcadm.sdc.metadata');
+
+            if (!self.sdcadm.sdc.metadata.fabric_cfg) {
+                err = new Error('cannot setup volapi: this DC is not setup ' +
+                    'for fabric networks');
+            }
+
+            next(err);
+        },
+
         function getVolApiPkg(ctx, next) {
             var filter = {name: svcData.params.package_name,
                 active: true};
